### PR TITLE
Command Launcher Update

### LINKED ISSUE #27: Bug when used for paths with spaces
This code does not work when the command contains a path with spaces:

https://github.com/Origen-SDK/o2/blob/c4662b5fb677b4e7d265e71e89f0a17fbd9687ab/rust/origen/src/core/os.rs#L24-L26

According to this comment using <code>cmd /C command_to_execute</code> argument is "normally" not needed. But, I suppose we have a case where it is needed.
https://stackoverflow.com/questions/44757893/cmd-c-doesnt-work-in-rust-when-command-includes-spaces

If I construct the return value exactly the same way as it's done for Linux, I don't have any issue. What led to the need for executing like this on Windows?

### PR DESCRIPTION
**Pull request by pderouen**
fixes #27 

@ginty recommended using <code>PathBuf</code> it cleaned up a lot of the issues I was seeing. I don't know that all of the added <code>OsString</code> is necessary. In my testing I had better luck when using <code>OsString</code>. In any case the string has to be converted to the appropriate type for the system eventually. So, no harm (in my view).

@coreyeng can you kick the tires a bit?

I think the os.rs file becomes basically unused in this branch. I left it in for now.

### CODE REVIEW

**Comment by coreyeng** [comment]
Thanks! This is working okay for me so far. If we're not using `os.rs` anymore then that can be removed.  The command handler there was really just to get something going on Windows.

**Comment by pderouen** [comment]
Sounds good. I will remove os.rs.

**Comment by pderouen** [comment]
<code>on_windows</code> and <code>on_linux</code> are being used by pyapi. So, I cleaned out the command launcher code instead of deleting os.rs.

**Comment by pderouen** [comment]
pushed an update that fixes #101 

Do the Travis builds test the CLI commands?

**Comment by ginty** [comment]
> Do the Travis builds test the CLI commands?

No, not yet

File: rust/origen/cli/src/commands/setup.rs [hunk 1]
```diff
@@ -64,23 +65,23 @@ pub fn run() {
...
 
             if poetry_version().unwrap().major != 1 {
                 // Have to use --preview here to get a 1.0.0 pre version, can only use versions for
                 // official releases
-                os::cmd(&PYTHON_CONFIG.poetry_command)
-                    .arg("self:update")
-                    .arg("--preview")
+                Command::new(&PYTHON_CONFIG.poetry_command)
+                    .arg(OsString::from("self:update"))
+                    .arg(OsString::from("--preview"))
```
Comment (ginty) [inline rust/origen/cli/src/commands/setup.rs:84]: Is this OsString creation really required here?
I think we have a lot of other places where we don't do this. We should either do this everywhere or nowhere I think.

**Comment by ginty** [comment]
If you wanted to add some CLI tests I would add a `cli_test.py` here - https://github.com/Origen-SDK/o2/tree/master/example/tests

You should be able to switch dirs out of the app directory in Python code if you wanted to test global commands too.

Note that I think the naming of `_test.py` in the file is important, as is naming the test functions `test_..`.

To invoke those tests run `poetry run pytest` for them all, or `poetry run pytest tests/cli_test.py`.
I think there are also ways to invoke specific tests only, though I'm not sure of the details.
The test framework is called pytest if you want to google it.

File: rust/origen/cli/src/commands/setup.rs (hunk 1 above)
Comment (pderouen) [reply rust/origen/cli/src/commands/setup.rs:84]: No, I'm not sure that it is needed here. The arg should be internally converting to OsString. I will back out the OsString changes and give it another try.

**Comment by pderouen** [comment]
> If you wanted to add some CLI tests I would add a `cli_test.py` here - https://github.com/Origen-SDK/o2/tree/master/example/tests

It was just an honest question, wondering if the linux compile of the cli gets the tires kicked. I don't know that it's necessary to directly test the cli. As we're working on updates it'll be getting a constant run through. I'm willing to take a stab at a few tests to at least check that it doesn't die on <code>origen v</code> or <code>origen g</code> if you'd like to see that.

**Comment by coreyeng** [comment]
Partially from #97 , and partially from the mess I've got [here](https://github.com/Origen-SDK/o2/blob/doc_system/rust/origen/cli/src/bin.rs#L326-L356), I think we'll need to rethink the command handler a bit before its all said and done. Might not be a bad idea to have some CLI tests to go off of. Not that it needs to be done now, just thinking out loud.

**Comment by pderouen** [comment]
Yeah, I think I changed my mind about tests for the CLI over night. It would be better to have tests for it. I'll put a little thought into it over the next day or 2, but I think I'd rather merge these changes as is.

**Comment by pderouen** [comment]
This seems like a good way to test the cli:
https://rust-cli.github.io/book/tutorial/testing.html#testing-cli-applications-by-running-them

**Comment by pderouen** [comment]
@coreyeng part of adding testing I was thinking to add for the cli _might_ involve moving the "do something with the args" code into a lib for direct unit testing... so instead of checking that <code>origen g</code> works when with a windows style path argument; I would just check that the strings passed to poetry are literals (and make that more likely by using a central fn or macro to build them).

Anyway, all of that to ask if you'd want to wait until after your branch is merged and I can work from there to add the testing? Or, what?

**Comment by pderouen** [comment]
I'm starting a cli_test branch to work on tests. I'll go ahead and merge this.

**Pull request merged**